src/vumi2/transports/smpp/client.py

```diff
@@ -111,7 +120,10 @@ async def enquire_link(self) -> None:
         while True:
             deadline = current_time() + self.config.smpp_enquire_link_interval
             pdu = EnquireLink(seqNum=await self.sequencer.get_next_sequence_number())
-            await self.send_pdu(pdu)
+            with move_on_after(self.config.smpp_enquire_link_interval) as cancel_scope:
+                await self.send_pdu(pdu)
+            if cancel_scope.cancel_called:
```

Comment:
According to the docs (https://trio.readthedocs.io/en/stable/reference-core.html#trio.CancelScope.cancelled_caught), `cancelled_caught` is more appropriate than `cancel_called`.